runtime/src/accounts.rs

```diff
@@ -858,25 +845,13 @@ impl Accounts {
     #[must_use]
     #[allow(clippy::needless_collect)]
     pub fn lock_accounts<'a>(&self, txs: impl Iterator<Item = &'a Transaction>) -> Vec<Result<()>> {
-        use solana_sdk::sanitize::Sanitize;
-        let keys: Vec<Result<_>> = txs
-            .map(|tx| {
-                tx.sanitize().map_err(TransactionError::from)?;
-
-                if Self::has_duplicates(&tx.message.account_keys) {
-                    return Err(TransactionError::AccountLoadedTwice);
-                }
-
-                Ok(tx.message().get_account_keys_by_lock_type())
-            })
+        let keys: Vec<_> = txs
+            .map(|tx| tx.message().get_account_keys_by_lock_type())
```

Comment:
not sure why the transactions were lazily sanitized just before locking accounts, but it makes good sense to sanitize them close to their entry point.